src/contract/basic_node_signer.ts

```diff
@@ -18,12 +24,64 @@ export const basicNodeSigner = (
   networkPassphrase: string,
 ) => ({
   // eslint-disable-next-line require-await
-  signTransaction: async (tx: string) => {
-    const t = TransactionBuilder.fromXDR(tx, networkPassphrase);
-    t.sign(keypair);
-    return t.toXDR();
+  signTransaction: async (
+    xdr: string,
+    opts?: {
+      networkPassphrase?: string;
+      address?: string;
+      submit?: boolean;
+      submitUrl?: string;
+    }
+  ): Promise<
+    | {
+        signedTxXdr: string;
+        signerAddress: string;
+      }
+    | {
+        error: WalletError;
+      }
+  > => {
```

Comment:
Why do we need to hardcode any of this? Can't we import these types from `@creit.tech/stellar-wallets-kit`?

Reply:
True, that was an oversight on my part. Although there is one small issue which is that the type in stellar-wallets-kit doesn't quite conform with the spec since it doesn't also potentially return the Error type. I think for the purposes of basicNodeSigner it doesn't matter but it might for the AssembledTransaction code.